js/coinsbit.js

```diff
@@ -0,0 +1,1114 @@
+        return this.safeOrder2 ({
+            'id': id,
+            'clientOrderId': tradeId, // <<<TODO>>> should it be here?
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'lastTradeTimestamp': undefined,
+            'status': undefined,
+            'symbol': symbol,
+            'type': orderType,
+            'timeInForce': undefined,
+            'side': side,
+            'price': price,
+            'stopPrice': undefined,
+            'average': undefined,
+            'amount': orderSize,
+            'cost': cost,
+            'filled': filledSize,
+            'remaining': remaining,
+            'fee': {
+                'cost': fee,
+                'currency': quoteCurrency,
+                'will_cost': willBeFee,
+            },
+            'trades': undefined,
+            'info': order,
+        }, market);
+    }
+
+    async fetchOrderTrades (id, symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {};
+        request['orderId'] = id;
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.privatePostAccountTrades (this.extend (request, params));
+        // {
+        //     success: true,
+        //     message: '',
+        //     result: {
+        //       offset: '0',
+        //       limit: '50',
+        //       records: {
+        //         time: '1640258718.402',
+        //         fee: '0.01572657',
+        //         price: '3931.64271609',
+        //         amount: '0.002',
+        //         id: '1595921',
+        //         dealOrderId: '8631729321',
+        //         role: '2',
+        //         deal: '7.86328543'
+        //       }
+        //     },
+        //     code: '200'
+        // }
+        const result = this.safeValue (response, 'result', {});
+        const order = this.safeValue (result, 'records');
+        if (!order) {
+            throw new OrderNotFound (this.id + ' fetchOrderTrades() could not find any trade data for order-id ' + id);
+        }
+        return this.parseTrade ({ 'id': id, 'order': order }, undefined);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const request = this.omit (params, 'type');
+        const currency = this.safeValue (params, 'currency');
```

Comment:
See how this is done in Deribit.

Reply:
I doubt I was unable to understand how deribit example is connected with this, as deribit has exceptionally different fetchBalance approach (using some another method, which in its turn, gets currency from hardcoded `this.options` properties).  However, If I correctly got the point, is that the user provided symbol should be checked against the 'currencies' property and from there assign the 'id'. If that's what you wanted, then yes sure. done .

Reply:
> However, If I correctly got the point, is that the user provided symbol should be checked against the 'currencies' property and from there assign the 'id'. If that's what you wanted, then yes sure. done .

The user should be able to supply one of two params: `currency` (exchange-specific id) and `code` (ccxt unified currency code). The former takes precedence over the latter and we should never mix the two.

```JavaScript
    async fetchBalance (params = {}) {
        await this.loadMarkets ();
        let currencyId = this.safeString (params, 'currency');
        if (currencyId === undefined) {
            const code = this.safeString (params, 'code');
            if (code !== undefined) {
                params = this.omit (params, 'code');
                const currency = this.currency (code);
                currencyId = currency['id'];
            }
        }
        let method = 'privatePostAccountBalances';
        if (currencyId !== undefined) {
            request['currency'] = currencyId;
            method = 'privatePostAccountBalance';
        }        
        const response = await this[method] (request);
        // ...
```